crates/polars-plan/src/plans/optimizer/projection_pushdown/joins.rs

```diff
@@ -19,16 +19,14 @@ fn add_keys_to_accumulated_state(
     // that means we don't want to execute the projection as that is already done by
     // the JOIN executor
     if add_local {
-        // take the left most name as output name
-        let mut iter = aexpr_to_leaf_names_iter(expr, expr_arena);
-        if let Some(name) = iter.next() {
-            drop(iter);
-            let node = expr_arena.add(AExpr::Column(name.clone()));
+        // return the left most name as output name
+        let names = aexpr_to_leaf_names_iter(expr, expr_arena).collect::<Vec<_>>();
```

Comment:
Instead of allocating a vec, can we do something like this?

```

let mut first = true;
let mut output_name = None;
for name in names {
    if first {
        output_name = Some(name.clone());
        first = false;
    }
    ...
}
```

Reply:
Unfortunately not because the `names` iterator holds a reference to expr_arena that we also want to use within the for loop. I don't see a trivial solution but can spend more time if it is important.

Reply:
Ah, right. :+1: